src/machines/activity_state_machine.rs

```diff
@@ -227,25 +213,49 @@ impl WFMachinesAdapter for ActivityMachine {
                 }),
             }
             .into()],
-            ActivityMachineCommand::Cancel(_) => unimplemented!(),
+            ActivityMachineCommand::Cancel => unimplemented!(),
+            ActivityMachineCommand::RequestCancellation(_) => unimplemented!(),
+        })
+    }
+}
+
+impl TryFrom<CommandType> for ActivityMachineEvents {
+    type Error = ();
+
+    fn try_from(c: CommandType) -> Result<Self, Self::Error> {
+        Ok(match c {
+            CommandType::ScheduleActivityTask => Self::CommandScheduleActivityTask,
+            CommandType::RequestCancelActivityTask => Self::CommandRequestCancelActivityTask,
+            _ => return Err(()),
         })
     }
 }
 
 impl Cancellable for ActivityMachine {
     fn cancel(&mut self) -> Result<MachineResponse, MachineError<Self::Error>> {
-        unimplemented!()
+        Ok(
+            match self.on_event_mut(ActivityMachineEvents::Cancel)?.pop() {
+                Some(ActivityMachineCommand::RequestCancellation(cmd)) => {
+                    MachineResponse::IssueNewCommand(cmd)
+                }
+                Some(ActivityMachineCommand::Cancel) => MachineResponse::NoOp,
```

Comment:
Probably best to leave this as `unimplemented!` rather than `NoOp` since the triggering command is unimplemented right now anyway. Might make for a confusing bit of debugging later.

Reply:
Ok